packages/password-input/src/PasswordInput/PasswordInput.tsx

```diff
@@ -0,0 +1,208 @@
+      autoComplete = 'new-password',
+      className,
+      label,
+      ...rest
+    }: PasswordInputProps,
+    forwardedRef,
+  ) => {
+    const [showPassword, setShowPassword] = useState<boolean>(false);
+    const prefix = 'lg-passwordinput';
+    const inputId = useIdAllocator({ prefix, id: idProp });
+    const descriptionId = useIdAllocator({
+      prefix,
+      id: ariaDescribedbyProp,
+    });
+    const labelId = useIdAllocator({
+      prefix,
+      id: ariaLabelledbyProp,
+    });
+    const { theme, darkMode } = useDarkMode(darkModeProp);
+    const { value, handleChange } = useControlledValue(valueProp, onChangeProp);
+
+    if (!label && !ariaLabelledbyProp && !rest['aria-label']) {
+      console.warn(
+        'For screen-reader accessibility, label, aria-labelledby, or aria-label must be provided',
+      );
+    }
+
+    // If disabled then hide password
+    useEffect(() => {
+      if (disabled) setShowPassword(false);
+    }, [disabled]);
+
+    const handleTogglePasswordClick = () => setShowPassword(s => !s);
+
+    /**
+     * The overall state of the component
+     */
+    const state: States = Array.isArray(stateNotifications)
+      ? getStateFromArray(stateNotifications)
+      : stateNotifications;
+
+    /**
+     * Checks if there are any notifications
+     */
+    const hasNotifications: boolean =
+      !ariaDescribedbyProp && Array.isArray(stateNotifications);
+
+    return (
+      <LeafyGreenProvider darkMode={darkMode}>
+        <div className={className} ref={forwardedRef}>
+          {label && (
+            <Label
+              id={labelId}
+              className={cx(labelBaseStyles, {
+                [labelLargeOverrideStyles]: sizeVariant === SizeVariant.Large,
+              })}
+              htmlFor={inputId}
+              disabled={disabled}
+            >
+              {label}
+            </Label>
+          )}
+          <div className={inputWrapperStyles}>
+            <input
+              type={showPassword ? 'text' : 'password'}
+              value={value}
+              id={inputId}
+              autoComplete={autoComplete}
+              aria-describedby={descriptionId}
+              aria-labelledby={labelId}
+              aria-disabled={disabled}
+              aria-invalid={state === States.Error || state === States.Warning}
+              className={cx(
+                inputBaseStyles,
+                inputSizeStyles[sizeVariant],
+                inputBaseThemeStyles[theme],
+                {
+                  [inputThemeStyles[theme][state]]: !disabled,
+                  [cx(
+                    inputDisabledBaseStyles,
+                    inputDisabledThemeStyles[theme],
+                  )]: disabled,
+                  [inputIconSizeStyles[sizeVariant]]:
+                    !hasNotifications && state !== States.None,
+                },
+              )}
+              onChange={handleChange}
+              readOnly={disabled ? true : false}
```

Comment:
use disabled

Reply:
i'm silly, it could also just be `disabled`

Reply:
but im gonna swap `readOnly` for `disabled`.

Reply:
Ah okay. I assumed this ternary was to ensure the value is never undefined, but that works too haha